### PR TITLE
feat: remove searcher context from harness and master [MD-498]

### PR DESCRIPTION
**Pull request by azhou-determined**
## Description

This is the feature branch of several previously reviewed and merged features:
- [Custom Searcher & DSAT Removal](https://github.com/determined-ai/determined/pull/9949): removes previously-deprecated custom searcher and DSAT features.
- [Remove WebUI usages of max_length](https://github.com/determined-ai/determined/pull/10038): removes WebUI dependencies on max_length
- [Refactor max_length from PyTorchTrial](https://github.com/determined-ai/determined/pull/10091): removes PyTorchTrial/Trainer dependencies on `max_length`
- [Introducing KerasCallback](https://github.com/determined-ai/determined/pull/10075): a new high-level training integration for TFKeras
- [HF Trainer refactor](https://github.com/determined-ai/determined/pull/10083): removes max_length and searcher dependencies from HF trainer
- [Refactor master-side searchers](https://github.com/determined-ai/determined/pull/10024): refactors master-side searchers to no longer depend on max_length
- [Add batches and epochs to TFKerasTrial](https://github.com/determined-ai/determined/pull/10129).
- [Update docs for changing special metric name from "epoch" to "epochs"](https://github.com/determined-ai/determined/pull/10130)
- [Introducing Deepspeed Trainer API](https://github.com/determined-ai/determined/pull/10100)
- [Finalize some deprecation warnings](https://github.com/determined-ai/determined/pull/10133)

In summary, this change removes the concept of searcher operations from all training. For details, please see the [design doc](https://hpe-aiatscale.atlassian.net/wiki/spaces/ENGINEERIN/pages/1684897828/Searcher+Context+Removal).

## Checklist

- [ ] Changes have been manually QA'd
- [ ] New features have been approved by the corresponding PM
- [ ] User-facing API changes have the "User-facing API Change" label
- [ ] Release notes have been added as a separate file under `docs/release-notes/`
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses have been included for new code which was copied and/or modified from any external code

[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by rb-determined-ai — APPROVED** [review]
I consider this review perfunctory, since the individual commits were reviewed, many of them very heavily.

**Review by davidfluck-hpe — APPROVED** [review]

**Review by highvelcty — APPROVED** [review]
The changes in the `performance/daist` directory lgtm. 

I do not have the context to comment on the other changes.

**Review by stoksc — APPROVED** [review]
lgtm on the premise that i already reviewed this code

**Review by gt2345 — APPROVED** [review]

**Review by MikhailKardash — APPROVED** [review]
Amazing wrok, LGTM!
